crates/router/src/core/payments/tokenization.rs

```diff
@@ -1252,3 +1280,42 @@ pub fn update_connector_mandate_details(
 
     Ok(connector_mandate_details)
 }
+
+#[cfg(feature = "v1")]
+pub fn update_connector_mandate_details_status(
+    merchant_connector_id: Option<id_type::MerchantConnectorAccountId>,
+    mandate_details: Option<diesel_models::PaymentsMandateReference>,
+) -> RouterResult<Option<serde_json::Value>> {
+    let mandate_reference = match mandate_details {
+        Some(mut payment_mandate_reference) => {
+            if let Some(mca_id) = merchant_connector_id {
+                payment_mandate_reference.entry(mca_id).and_modify(|pm| {
+                    let update_rec = diesel_models::PaymentsMandateReferenceRecord {
+                        connector_mandate_id: pm.connector_mandate_id.clone(),
```

Comment:
you can just update the `connector_mandate_status` field only right?

Reply:
Yea the others needs to be the same